range-v3/include/range/v3/utility/associated_types.hpp

```diff
@@ -31,7 +31,9 @@ namespace ranges
         ////////////////////////////////////////////////////////////////////////////////////////
         template<typename T, typename Enable /*= void*/>
         struct difference_type
-        {};
+        {
+          using type = std::ptrdiff_t;
+        };
```

Comment:
where is this used?

Reply:
I get this otherwise.
```
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/dot_product_zip_copy.cpp:1:
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/../../gstorm/include/gstorm.h:8:
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/../../gstorm/include/detail/operators/copy.h:7:
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/../../gstorm/include/detail/traits.h:7:
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/../../gstorm/include/detail/ranges/vector.h:16:
In file included from /home/toomas/projects/parallelts-range-exploration/examples/tryout/../../range-v3/include/range/v3/view_facade.hpp:18:
/home/toomas/projects/parallelts-range-exploration/examples/tryout/../../range-v3/include/meta/meta.hpp:140:9: error:
      no type named 'type' in 'ranges::v3::difference_type<meta::v1::nil_, void>'
        using _t = typename T::type;
        ^~~~~
/home/toomas/projects/parallelts-range-exploration/examples/tryout/../../range-v3/include/meta/meta.hpp:140:9: error:
      no type named 'type' in 'ranges::v3::difference_type<meta::v1::nil_, void>'
        using _t = typename T::type;
        ^~~~~
/home/toomas/projects/parallelts-range-exploration/examples/tryout/../../range-v3/include/meta/meta.hpp:140:9: error:
      no type named 'type' in 'ranges::v3::difference_type<meta::v1::nil_, void>'
        using _t = typename T::type;
        ^~~~~
/home/toomas/projects/parallelts-range-exploration/examples/tryout/../../range-v3/include/meta/meta.hpp:140:9: error:
      no type named 'type' in 'ranges::v3::difference_type<meta::v1::nil_, void>'
        using _t = typename T::type;
   ```

Reply:
I get this only with SYCL though.